File: Java/EX6.java
// Classe Motor
class Motor {
    private int potencia; // em cavalos de potência (cv)
    private String tipoCombustivel; // gasolina, etanol, etc.

    // Construtor
    public Motor(int potencia, String tipoCombustivel) {
        this.potencia = potencia;
        this.tipoCombustivel = tipoCombustivel;
    }

    // Método para exibir informações do motor
    public void exibirInformacoes() {
        System.out.println("Motor: " + potencia + " cv, Tipo de Combustível: " + tipoCombustivel);
    }
}

// Classe Carro
class Carro2 {
    private String marca;
    private String modelo;
    private int ano;
    private Motor motor; // Atributo que é um objeto da classe Motor

    // Construtor
    public Carro2(String marca, String modelo, int ano, Motor motor) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.motor = motor;
    }

    // Método para exibir detalhes do carro
    public void exibirDetalhes() {
        System.out.println("Marca: " + marca + ", Modelo: " + modelo + ", Ano: " + ano);
        motor.exibirInformacoes(); // Exibe informações do motor associado
    }
}

// Classe principal
class Main2 {
    public static void main(String[] args) {
        // Criando um objeto Motor
        Motor motor1 = new Motor(150, "Gasolina");

        // Criando um objeto Carro associado ao Motor
        Carro2 carro1 = new Carro2("Toyota", "Yaris", 2023, motor1);

        // Exibindo detalhes do carro e do motor
        carro1.exibirDetalhes();
    }
}

File: Go/EX6.go
package main

import "fmt"

// Estrutura Motor
type Motor struct {
	potencia         int    // em cavalos de potência (cv)
	tipoCombustivel  string // gasolina, etanol, etc.
}

// Método para exibir informações do motor
func (m *Motor) exibirInformacoes() {
	fmt.Printf("Motor: %d cv, Tipo de Combustível: %s\n", m.potencia, m.tipoCombustivel)
}

// Estrutura Carro
type Carro struct {
	marca   string
	modelo  string
	ano     int
	motor   Motor // Atributo que é um objeto da estrutura Motor
}

// Método para exibir detalhes do carro
func (c *Carro) exibirDetalhes() {
	fmt.Printf("Marca: %s, Modelo: %s, Ano: %d\n", c.marca, c.modelo, c.ano)
	c.motor.exibirInformacoes() // Exibe informações do motor associado
}

// Função principal
func main() {
	// Criando um objeto Motor
	motor1 := Motor{potencia: 150, tipoCombustivel: "Gasolina"}

	// Criando um objeto Carro associado ao Motor
	carro1 := Carro{marca: "Toyota", modelo: "Yaris", ano: 2023, motor: motor1}

	// Exibindo detalhes do carro e do motor
	carro1.exibirDetalhes()
}
